staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/build_controller.go

```diff
@@ -366,6 +379,44 @@ func (bc *BuildController) setBuildDefaultProxy(spec *configv1.ProxySpec) {
 	bc.buildDefaults.DefaultProxy = spec
 }
 
+func (bc *BuildController) setBuildGitProxy(spec *configv1.ProxySpec) {
+	bc.configLock.Lock()
+	defer bc.configLock.Unlock()
+	if bc.buildDefaults.Config == nil {
+		bc.buildDefaults.Config = &openshiftcontrolplanev1.BuildDefaultsConfig{}
+	}
+	bc.buildDefaults.Config.GitHTTPProxy = spec.HTTPProxy
+	bc.buildDefaults.Config.GitHTTPSProxy = spec.HTTPSProxy
+	bc.buildDefaults.Config.GitNoProxy = spec.NoProxy
+}
+
+func (bc *BuildController) setBuildDefaultProxyFromProxyStatus(proxy *configv1.Proxy) {
+	bc.configLock.Lock()
+	defer bc.configLock.Unlock()
+	defaultProxy := bc.buildDefaults.DefaultProxy
+	if defaultProxy == nil {
+		defaultProxy = &configv1.ProxySpec{}
+		bc.buildDefaults.DefaultProxy = defaultProxy
+	}
+	defaultProxy.HTTPProxy = proxy.Status.HTTPProxy
+	defaultProxy.HTTPSProxy = proxy.Status.HTTPSProxy
+	defaultProxy.NoProxy = proxy.Status.NoProxy
```

Comment:
Now that it's coming together, I'm worried about data races if builds only have a proxy for pulling images (no global proxy config).
We could have a situation during an update where the proxy data is temporarily wiped out, only to have a build then start with invalid proxy configurations.

Reply:
interesting

so that would imply needing to examine
- the new global proxy settings
- and pre-existing build.config obj settings 

in conjunction under the lock, where we still give the pre-existing build.config obj settings precedence

OK I'll start working on an update

Reply:
correct.

Reply:
update pushed @adambkaplan PTAL